### ISSUE TITLE
incorrect file extension with dialog.showSaveDialog when `filters` is used

### ISSUE THREAD

**Issue by gch1p** [electron/electron#9455]
* Electron version: 1.6.7, 1.7.0, maybe more
* Operating system: macOS

See this code (example):
```
      const dialog = require('electron').remote.dialog
      dialog.showSaveDialog(require('electron').remote.getCurrentWindow(), {
        defaultPath: '/tmp/test.jpg',
        filters: [
          { name: 'Images', extensions: ['jpg', 'jpeg', 'png', 'gif'] }
        ]
      }, function(filename) {})
```

Note that the filename should be `test.jpg`. This is what happens in reality:
<img width="707" alt="2017-05-13 15 53 21" src="https://cloud.githubusercontent.com/assets/18737999/26025632/745f1a52-37f4-11e7-8a28-091c9a2a2cdb.png">

But if I remove the `filters` field, then it's correct:
<img width="724" alt="2017-05-13 15 53 33" src="https://cloud.githubusercontent.com/assets/18737999/26025633/8118e962-37f4-11e7-8a3a-c576a91f44d8.png">

**Comment by shubham2892** [comment]
Did a little digging and seems like on [[dialog setAllowedFileTypes:file_types]](https://github.com/electron/electron/blob/master/atom/browser/ui/file_dialog_mac.mm#L43),  the extension in the file name is changed to first _extension type_ passed in `file_types` variable which happened to be `png` in above example.

**Comment by kevinsawicki** [comment]
Fixed by @shubham2892 in #9610

**Comment by pravindia** [comment]
https://stackoverflow.com/questions/55074860/unable-to-add-filters-to-showsavedialog-in-electron
